### PR TITLE
New - trigger online sign

### PR DESCRIPTION
**Pull request by Brice-VINCENT**
# NEW|New [*Trigger Online Sign*]
[*Tool to send an email when signing an online quote.*]

### CODE REVIEW

File: htdocs/core/ajax/onlineSign.php [hunk 1]
```diff
@@ -200,6 +200,16 @@
 
 				dol_syslog(__METHOD__, LOG_DEBUG);
 				$resql = $db->query($sql);
+				//Trigger propal signe en ligne
+				if (method_exists($object, 'call_trigger')) {
+					$user = new User($db);
+					$user->fetch($object->user_valid_id);
+					$result = $object->call_trigger('PROPAL_CLOSE_SIGNED_ONLINE', $user);
```
Comment (eldy) [inline htdocs/core/ajax/onlineSign.php:207]: There is no need to duplicate the trigger PROPAL_CLOSE_SIGNED into a variant PROPAL_CLOSE_SIGNED_ONLINE.
What you can do is just add the line
`$object->context = array('closedfromonlinesignature' => 'closedfromonlinesignature')`
before the 
$result = $object->call_trigger('PROPAL_CLOSE_SIGNED', $user); line 227

File: htdocs/public/onlinesign/newonlinesign.php [hunk 2]
```diff
@@ -193,6 +193,15 @@
 	}
 
 	$object->fetch(0, $ref);
+	//Trigger propal signe en ligne
```
Comment (eldy) [inline htdocs/public/onlinesign/newonlinesign.php:196]: I pushed a commit to add the context on $object so this trigger call is useless.

File: htdocs/core/triggers/interface_70_modpropale_MailPropalSigne.class.php [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ *  \file       htdocs/core/triggers/interface_70_modpropale.MailPropalSigne.class.php
```
Comment (eldy) [inline htdocs/core/triggers/interface_70_modpropale_MailPropalSigne.class.php:19]: Rename your trigger interface_50_modPropale_Mail.class.php and use the trigger code PROPAL_CLOSE_SIGNED
and send email when $object->context contains the key closedfromonlinesignature

**Comment by BB2A-Anthony** [comment]
Hi its good job 

if the proposal is refused it also works?

**Comment by eldy** [comment]
> Hi its good job
> 
> if the proposal is refused it also works?

I think so, the context has also been added when trigger PROPAL_CLOSE_REFUSED is called
```
		$object->context = array('closedfromonlinesignature' => 'closedfromonlinesignature');
		$result = $object->call_trigger('PROPAL_CLOSE_REFUSED', $user);

```

**Comment by eldy** [comment]
An other solution was implemented differently so PR is closed.

**Comment by BB2A-Anthony** [comment]
What the solution

**Pull request closed without merging**